.github/workflows/build-pyodide-package.yml

```diff
@@ -0,0 +1,108 @@
+name: Build pyodide arbor
+
+on:
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    env:
+      PYODIDE_VERSION: 0.22.0a3
+      PYODIDE_RELEASE_URL: "https://github.com/pyodide/pyodide/releases/download"
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        submodules: true
+        path: arbor_src
+    - name: get-xbuildenv
+      run: |
+        XBUILDENV_BASENAME=xbuildenv-${PYODIDE_VERSION}.tar.bz2
+        wget -nc "${PYODIDE_RELEASE_URL}/${PYODIDE_VERSION}/${XBUILDENV_BASENAME}"
+        tar -xvf ${XBUILDENV_BASENAME}
+        rm ${XBUILDENV_BASENAME}
+    - name: read-makefile
+      run: |
+        make -f xbuildenv/pyodide-root/Makefile.envs .output_vars | grep 'PY' | tee -a $GITHUB_ENV
+    - name: get-emsdk
+      run: |
+        wget https://github.com/emscripten-core/emsdk/archive/refs/heads/main.zip
+        unzip main.zip
+        rm main.zip
+    - name: setup-emsdk
+      run: |
+        ./emsdk-main/emsdk install ${{env.PYODIDE_EMSCRIPTEN_VERSION}}
+        ./emsdk-main/emsdk activate ${{env.PYODIDE_EMSCRIPTEN_VERSION}}
+    - name: build-modcc-host
+      run: |
+        mkdir -p arbor_src/build_modcc_host
+        cd arbor_src/build_modcc_host
+        cmake .. -DARB_USE_BUNDLED_LIBS=ON
+        make modcc -j 8
+    - name: build-libarbor-emscripten
+      run: |
+        ABS_TOOLCHAIN=$(realpath emsdk-main/upstream/emscripten/cmake/Modules/Platform/Emscripten.cmake)
+        ABS_MODCC=$(realpath arbor_src/build_modcc_host/bin/modcc)
+        sed -i.bak -E \
+            '/only been tested|nonexistent_file|unbalanced brace/d' \
+            arbor_src/ext/random123/include/Random123/features/gccfeatures.h
+        mkdir -p arbor_src/build_arborlib_em
+        cd arbor_src/build_arborlib_em
+        cmake .. \
+            --toolchain "${ABS_TOOLCHAIN}" \
+            -DCMAKE_INSTALL_PREFIX=$(realpath prefix) \
+            -DARB_USE_BUNDLED_LIBS=ON \
+            -DARB_MODCC="${ABS_MODCC}" \
+            -DARB_ARCH=generic
+        make -j 8
+        make install
+    - name: build-arbor-python-so-emscripten
+      run: |
+        ABS_XBUILDENV_INCLUDE=$(realpath xbuildenv/pyodide-root${{ env.CPYTHONINSTALL }}/include/python${{ env.PYMAJOR }}.${{ env.PYMINOR }})
+        ABS_EM_ARBOR=$(realpath arbor_src/build_arborlib_em/prefix)
+        PY_VERSION_FLAT=${{ env.PYMAJOR }}${{ env.PYMINOR }}
+        ARBOR_PYLIB_OUTPUT=arbor/_arbor.cpython-${PY_VERSION_FLAT}-wasm32-emscripten.so
+        ABS_ARBOR_PYLIB_OUTPUT=$(pwd)/$ARBOR_PYLIB_OUTPUT
+        source emsdk-main/emsdk_env.sh
+        mkdir -p arbor arbor.dist-info
+        mkdir -p arbor_src/build_python_lib_em
+        cd arbor_src/build_python_lib_em
+        time em++ \
+            ../python/*.cpp \
+            -I ../ext/pybind11/include/ \
```

Comment:
Are all these flags required?

Reply:
Sadly yes. Usually these are produced by CMake or the actual build tool, but I could not convince CMake to produce an pyodide/emscripten .so file. About half the flags are required by arbor and make sure it find the right python header files, the other half is needed to make the .so file `import`-able by pyodide